sources/openeuler-http.go

```diff
@@ -0,0 +1,122 @@
+package sources
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"net/url"
+	"path/filepath"
+
+	"github.com/lxc/distrobuilder/shared"
+)
+
+type openEuler struct {
+	commonRHEL
+	fileName     string
+	checksumFile string
+}
+
+const (
+	isoFileName = "%s-%s-dvd.iso"
+	shaFileName = "%s-%s-dvd.iso.sha256sum"
+)
+
+func (s *openEuler) Run() error {
+	var err error
+	baseURL := fmt.Sprintf("%s/%s/ISO/%s/", s.definition.Source.URL,
+		s.definition.Image.Release,
+		s.definition.Image.Architecture)
+
+	fpath := s.getTargetDir()
+
+	s.fileName = fmt.Sprintf(isoFileName, s.definition.Image.Name, s.definition.Image.Architecture)
+	s.checksumFile = fmt.Sprintf(shaFileName, s.definition.Image.Name, s.definition.Image.Architecture)
+
+	_, err = url.Parse(baseURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse URL %s", baseURL)
```

Comment:
Please capitalize error messages and always include the actual error if present:

```go
return fmt.Errorf("Failed to parse URL %s: %w", baseURL, err)
```

Reply:
If capitalize the error messages, linter compalined as [error strings should not be capitalized (ST1005)go-staticcheck](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), but also notice all the error string style fellow the captialize style, if this style is distrobuiler followed, comment me and i will 'fix' it.

Reply:
We ignore ST1005 (see Makefile), and capitalize the error messages.

Reply:
Okay